guides/integrate-nextauth.md

```diff
@@ -0,0 +1,427 @@
+---
+title: NextAuth
+sidebar_position: 6
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import SetupJavaScript from '/docs/workflows/_sdk-setup/_setup-javascript.mdx';
+import Arcade, {Clip} from '/src/components/Arcade.tsx';
+
+# Integrate Beyond Identity Passwordless Authentication into NextAuth
+
+This guide provides information on how to set up Beyond Identity as a passwordless authentication provider for a [Next](https://nextjs.org) application that uses [NextAuth](https://next-auth.js.org/) (which is becoming [Auth.js](https://authjs.dev)).
+
+This guide will cover:
+
+- How to install the JavaScript SDK
+- How to configure Beyond Identity as an Identity Provider
+- How to create an identity and generate a passkey
+- How to authenticate with a passkey
+- How to configure the OIDC NextAuth plugin
+
+## Prerequisites
+
+- Set up a [developer account](/docs/v1/workflows/account-setup)
+
+## NextAuth
+
+- [NextAuth.js Initialization](https://next-auth.js.org/configuration/initialization)
+- [NextAuth.js OAuth Providers](https://next-auth.js.org/configuration/providers/oauth)
+
+- [Auth.js Introduction](https://authjs.dev/getting-started/introduction)
+- [Auth.js Guides](https://authjs.dev/guides)
+- [Beyond Identity Provider](https://authjs.dev/reference/core/providers_beyondidentity)
+
+### Example
+
+<Tabs groupId="nextjs" queryString>
+<TabItem value="nextauth" label="NextAuth.js">
+
+```javascript
+import NextAuth from "next-auth"
+
+export default NextAuth({
+  providers: [
+    {
+      id: "beyondidentity",
+      name: "Beyond Identity",
+      type: "oauth",
+      wellKnown: `https://auth-${REGION}.beyondidentity.com/v1/tenants/${TENANT_ID}/realms/${REALM_ID}/applications/${APPLICATION_ID}/.well-known/openid-configuration`,
+      authorization: { params: { scope: "openid" } },
+      clientId: process.env.APP_CLIENT_ID,
+      clientSecret: process.env.APP_CLIENT_SECRET,
+      idToken: true,
+      checks: ["state"],
+      profile(profile) {
+        return {
+          id: profile.sub,
+          name: profile.sub,
+          email: profile.sub,
+        }
+      }
+    }
+  ]
+})
+```
+
+</TabItem>
+<TabItem value="auth" label="Auth.js">
+
+```javascript
+import { Auth } from "@auth/core"
+import BeyondIdentity from "@auth/core/providers/beyondidentity"
+
+const request = new Request("https://example.com")
+const response = await Auth(request, {
+  providers: [BeyondIdentity({ clientId: "", clientSecret: "", issuer: "" })],
+})
+```
+
+</TabItem>
+</Tabs>
+
+We will overwrite the `Request` _url_ and `BeyondIdentity` _clientId_, _clientSecret_ and _issuer_ values later in this process, so you can use the dummy value for now.
```

Comment:
It might make sense to move this into the Auth.js tab so that it only shows when you tap that tab.

Reply:
Makes sense to me. Done.